mlrun/datastore/base.py

```diff
@@ -262,6 +471,7 @@ def as_df(
         file_url = self._sanitize_url(url)
         is_csv, is_json, drop_time_column = False, False, False
         file_system = self.filesystem
+        create_partition_path = kwargs.pop("create_partition_path", True)
```

Comment:
Parameter name could be much clearer (`optimize_partition_discovery`?). A comment might be helpful as well.